tests/access-control.tests.ts

```diff
@@ -57,12 +61,39 @@ describe('MY_ACCESS_CONTROL', () => {
 
     // Act - Alice transfer the token form sender to bob
     await expect(fromSigner(contract, alice.address).tx.transferFrom(sender.address, bob.address, bnArg(0)))
-        .to.eventually.be.fulfilled
+      .to.eventually.be.fulfilled
 
     // Assert - Bob is now owner of the token
     await expect(query.ownerOf(bnArg(0))).to.have.output(bob.address)
   })
 
+  it('PSP 721 - safe transfer works', async () => {
```

Comment:
Will be cool to test the scenario where the Receiver will reject the transfer to verify that it works properly.

For that, you can add an additional method to mock contract `revert_next_transfer()`